### DISCUSSION TITLE
Mapping an array of objects into a dp.Group

### DISCUSSION THREAD

**Discussion by andrewhong5297** [datapane/datapane#138] [Ideas]
Hey all! not sure if there's an example somewhere of how to map tables into dp.Group, say something like this :

```
tables = [fig1,fig2,fig3]

dp.Group(
for table in tables:
      dp.Select(table)
columns=len(tables)
)
```

**Comment by andrewhong5297** [comment]
obvi know I can't for loop inside the Group, but doing it outside and storing select objects doesn't seem to work for me either

**Comment by mands** [comment]
Hi there,

So there are a couple of ways to do this, probably the easiest would be a list comprehension, e.g.

```python
# map a new list of Selects where each Select contains a single table
selects = [dp.Select(table) for table in tables]

# then, can add the selects to the group, using `*` to expand the list, e.g.
dp.Group(*selects, columns=len(selects))

# OR, using the blocks parmeter, which takes a list of blocks
dp.Group(blocks=selects, columns=len(selects))
```

Note, `dp.Select` works best with at least 2 blocks per select, so the user can select/choose between them - but you would structure such a array directly in the same way.

HTH!
